datacompy/spark.py

```diff
@@ -146,9 +146,9 @@ class SparkCompare:
 
     def __init__(
         self,
-        spark_session: SparkSessionType,
-        base_df: SparkDataFrame,
-        compare_df: SparkDataFrame,
+        spark_session: "SparkSessionType",
```

Comment:
@aguiddir these might need to be "pyspark.sql.SparkSession" and "pyspark.sql.DataFrame" since "SparkSession" and "DataFrame" wouldn't be defined for mypy.

You might also be able to remove 
```python
    SparkSessionType = pyspark.sql.SparkSession
    SparkDataFrame = pyspark.sql.DataFrame
```
from the `try`

Reply:
@fdosani Done :+1:

Reply:
Thank you! Just one more question. Runing mypy gives me this: 
```
datacompy/spark.py:22: error: Cannot find implementation or library stub for module named "pyspark"  [import-not-found]
datacompy/spark.py:23: error: Cannot find implementation or library stub for module named "pyspark.sql.functions"  [import-not-found]
datacompy/spark.py:23: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
datacompy/spark.py:23: error: Cannot find implementation or library stub for module named "pyspark.sql"  [import-not-found]
datacompy/spark.py:463: error: Unused "type: ignore" comment  [unused-ignore]
datacompy/spark.py:472: error: Unused "type: ignore" comment  [unused-ignore]
```

Do we need `type: ignore` on these lines (463 and 472)?

Reply:
@fdosani Yes, i think we need it because the `orderby` method takes the argument: `*cols: Union[str, Column, List[Union[str, Column]]]`.

The argument we're passing to the method:` self._join_column_names` is of type: `List[str]`.
If we don't address the error, mypy returns this error message:
 `Argument 1 has incompatible type "List[str]"; expected "Union[str, Column, List[Union[str, Column]]]" [arg-type]`

which, in my opinion, isn't a valid error.

I think that to fix this issue, the modification should be made in the 'pyspark' library. The signature of the '[sort](https://github.com/apache/spark/blob/e359318c4493e16a7546d70c9340ffc5015aacff/python/pyspark/sql/dataframe.py#L3036)' method  should change from  `*cols: Union[str, Column, List[Union[str, Column]]]` to ->  `*cols: Union[str, Column, Sequence[Union[str, Column]]]`.

Reply:
Sounds good. Thanks for all you rhelp and patience with these changes etc.

Reply:
Here is the mypy [documentation](https://mypy.readthedocs.io/en/stable/common_issues.html#variance) that explains a bit more about where our problem is coming from.  We can also try to make corrections on our side, but wouldn't it be better to use` # type: ignore[arg-type]` ?